### PR TITLE
Couch to SQL migration: CommtrackConfig (1 of 3)

### PR DESCRIPTION
**Pull request by orangejenny**
[Docs](https://github.com/dimagi/commcare-hq/blob/master/docs/couch_to_sql_models.rst)

This one is fairly long because `CommtrackSettings` has five sub-models. To stick with the original design, each of these is turned into a separate table, either a `ForeignKey` or a `OneToOneField`. I initially wrote more generic syncing code for handling one to one fields and ultimately deleted it because there aren't that many models that would use it.

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-1323)

### CODE REVIEW

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Comment by orangejenny** [comment]
Rebased and force pushed because I pulled in a commit from the next PR by accident and wanted to remove it.

The only genuine change is the addition of https://github.com/dimagi/commcare-hq/pull/27597/commits/d9014c11ce8fbe5de85ae95bf19f6f146b355f03 which adds diff methods.

**Review by millerdev — APPROVED** [review]
Thanks for noting the new parts

**Comment by orangejenny** [comment]
Force pushes don't affect any of the old code, I just wanted to one of the most recent commits shortly after pushing it.

File: corehq/apps/cleanup/tests/test_couch_to_sql_commtrack.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def _assert_sql(self, sql, use_auto_consumption=False, min_window=2, first_keywo
         self.assertEqual(sql.domain, "my_project")
         self.assertFalse(sql.use_auto_emergency_levels)
         self.assertFalse(sql.sync_consumption_fixtures)
-        self.assertFalse(use_auto_consumption)
+        self.assertTrue(sql.use_auto_consumption == use_auto_consumption)
```
Comment (millerdev) [inline corehq/apps/cleanup/tests/test_couch_to_sql_commtrack.py:161]: nit
```suggestion
        self.assertEqual(sql.use_auto_consumption, use_auto_consumption)
```

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Pull request merged**